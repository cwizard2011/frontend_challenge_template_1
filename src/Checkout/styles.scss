.form-container-billing {
  width: 40%;
  display: flex;
  flex-direction: column;
  margin: auto;
  margin-top: 30px;
}

.billing-text {
  text-align: center;
  font-family: 'Montserrat';
  font-size: 25px;
}

.stripe-checkout-button-disabled {
  pointer-events: none;
  & button {
    background: none !important;

    & span {
      background-image: none !important;
      background-color: gray !important;
      cursor: not-allowed;
    }
  }
}