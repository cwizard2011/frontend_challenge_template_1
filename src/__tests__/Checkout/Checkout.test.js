import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Checkout } from '../../Checkout';

describe('<Checkout />', () => {
  const props = {
    history: {
      location: {
        pathname: ''
      }
    }
  }
  const wrapper = mount(<Checkout {...props} />);

  it('should render without crashing', () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should update the user\'s address information', () => {
    wrapper.setState({ isLoading: false });
    const value = "No 1, Birmingham Street";
    wrapper.find('#address_1').at(0).simulate('change', {
      target: {
        name: 'address_1',
        value
      }
    });
    const newUser = { address_1: value }
    expect(wrapper.state('userObj')).toEqual(newUser);
  });
});
