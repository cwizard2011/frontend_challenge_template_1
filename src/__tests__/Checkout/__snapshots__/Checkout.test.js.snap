// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkout /> should render without crashing 1`] = `
<Checkout
  history={
    Object {
      "location": Object {
        "pathname": "",
      },
    }
  }
>
  <Spinner>
    <div
      className="spinner"
    />
  </Spinner>
</Checkout>
`;
